gensim/parsing/preprocessing.py

```diff
@@ -125,17 +317,84 @@ def stem_text(text):
 
 
 def preprocess_string(s, filters=DEFAULT_FILTERS):
+    """Takes string, applies list of chosen filters to it, where filters are methods from this module. Default list of filters consists of: strip_tags, strip_punctuation, strip_multiple_whitespaces, strip_numeric, remove_stopwords, strip_short, stem_text. <function <lambda>> in signature means that we use lambda function for applying methods to filters.
+
+    Parameters
+    ----------
+    s : str
+    filters: list, optional
+
+    Returns
+    -------
+    list
+        List of unicode strings.
+
+    Examples
+    --------
+    >>> from gensim.parsing.preprocessing import preprocess_string
+    >>> s = "<i>Hel 9lo</i> <b>Wo9 rld</b>! Th3     weather_is really g00d today, isn't it?"
+    >>> preprocess_string(s)
+    [u'hel', u'rld', u'weather', u'todai', u'isn']
+
+    >>> from gensim.parsing.preprocessing import preprocess_string, strip_tags, strip_punctuation
+    >>> s = "<i>Hel 9lo</i> <b>Wo9 rld</b>! Th3     weather_is really g00d today, isn't it?"
+    >>> CUSTOM_FILTERS = [lambda x: x.lower(), strip_tags, strip_punctuation]
+    >>> preprocess_string(s,CUSTOM_FILTERS)
+    [u'hel', u'9lo', u'wo9', u'rld', u'th3', u'weather', u'is', u'really', u'g00d', u'today', u'isn', u't', u'it']
+
+    """
+
     s = utils.to_unicode(s)
     for f in filters:
         s = f(s)
     return s.split()
 
 
 def preprocess_documents(docs):
+    """Takes list of strings, splits it into sentences, then applies default filters to every sentence.
+
+    Parameters
+    ----------
+    docs : list
+
+    Returns
+    -------
+    list
+        List of lists, filled by unicode strings.
+
+    Examples
+    --------
+    >>> from gensim.parsing.preprocessing import preprocess_documents
+    >>> s = ["<i>Hel 9lo</i> <b>Wo9 rld</b>!", "Th3     weather_is really g00d today, isn't it?"]
+    >>> preprocess_documents(s)
+    [[u'hel', u'rld'], [u'weather', u'todai', u'isn']]
+
+    """
+
     return [preprocess_string(d) for d in docs]
 
 
 def read_file(path):
+    r"""Reads file in specified directory.
+
+    Parameters
+    ----------
+    path : str
+
+    Returns
+    -------
+    list
+        List of unicode strings.
+
+    Examples
+    --------
+    >>> from gensim.parsing.preprocessing import read_file
+    >>> path = "/media/work/october_2017/gensim/gensim/test/test_data/mihalcea_tarau.summ.txt"
```

Comment:
This path will works only on your filesystem, utils to retrieve path to test files will be ready very soon